01. The data in *.fq format was aligned with the Nipponbare genome or the 20 kb of Sa locus (included upstream and downstream sequence of Sa, ref.fa) and the output files were gvcf and vcf.
02. Five key SNPs were extracted from gvcf, corresponding to 6435/11236/16183/16316 position in ref.fa, and 22376434/22381235/22386181/22386182/22386315 position in the Nipponbare genome.
03. Convert hap.txt to +/-, '+' means the indica type while '-' means the japonica type.
04. Calculate missing gap information of SaM.
